plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -554,12 +556,17 @@ SecRule REQUEST_FILENAME "@rx /apps/text/(?:public/)?(?:session/(?:sync|close|pu
     nolog,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
     chain"
-    SecRule ARGS:json.sessionToken "@rx ^(?i)[a-z0-9+/]+$" \
+    SecRule ARGS:sessionToken|ARGS:json.sessionToken "@rx ^(?i)[a-z0-9+/]+$" \
         "t:none,\
+        ctl:ruleRemoveTargetById=920273;ARGS:sessionToken,\
+        ctl:ruleRemoveTargetById=932236;ARGS:sessionToken,\
+        ctl:ruleRemoveTargetById=942432;ARGS:sessionToken,\
+        ctl:ruleRemoveTargetById=942450;ARGS:sessionToken,\
         ctl:ruleRemoveTargetById=920273;ARGS:json.sessionToken,\
         ctl:ruleRemoveTargetById=932236;ARGS:json.sessionToken,\
         ctl:ruleRemoveTargetById=942432;ARGS:json.sessionToken,\
-        ctl:ruleRemoveTargetById=942450;ARGS:json.sessionToken"
+        ctl:ruleRemoveTargetById=942450;ARGS:json.sessionToken,\
+        ctl:ruleRemoveTargetById=920273;REQUEST_BODY"
```

Comment:
Do we really want to exclude this only if `sessionToken` matches regex above?

Reply:
The idea was to use character whitelisting to make the rule exclusion stricter.

Reply:
I understand that part but whitelisting `REQUEST_BODY` only if some header matches some regex doesn't make sense to me. Why does `REQUEST_BODY` matches rule `920273` only if `sessionToken` matches that regex? How is that connected?

Reply:
920273 checks `ARGS` and `REQUEST_BODY`. So we only need to exclude the request body in this particular case.

Reply:
> 920273 checks `ARGS` and `REQUEST_BODY`. So we only need to exclude the request body in this particular case.

Unfortunately it doesn't work like that. We (may, it depends) need both for correct exclusion.

Reply:
I'm marking as resolved to not block merging this PR.

Reply:
You mean because there may be other cases where we need to exclude the body? Sorry, I can't follow your logic.

Reply:
I want to cut release 1.1.0 of the plugin, so if you want to change something before that, let me know.

Reply:
> You mean because there may be other cases where we need to exclude the body? Sorry, I can't follow your logic.

Because we are excluding for specific arguments / variables. If you exclude a rule for `REQUEST_BODY` it won't stop mod_security to apply that rule to other variables.

Reply:
> I want to cut release 1.1.0 of the plugin, so if you want to change something before that, let me know.

No, go ahead!

Reply:
> Because we are excluding for specific arguments / variables. If you exclude a rule for REQUEST_BODY it won't stop mod_security to apply that rule to other variables.

I understand that. Well, let's discuss it when it comes up again :)